Dapper/CommandDefinition.cs

```diff
@@ -126,6 +131,8 @@ internal IDbCommand SetupCommand(IDbConnection cnn, Action<IDbCommand, object> p
             if (CommandType.HasValue)
                 cmd.CommandType = CommandType.Value;
             paramReader?.Invoke(cmd, Parameters);
+            if (RegisterCancellation)
+                CancellationToken.Register(() => cmd.Cancel()); 

```

Comment:
This should be `cancellationToken.Register(static state => ((IDbCommand)state).Cancel(), cmd);` to avoid a capture context and delegate allocation, however:

1. honestly, I think we should pause on this a sec while we discuss the overall strategy and API intent
2. we'd also need to think about how to handle the `CancellationTokenRegistration` (return value)

Reply:
Acknowledged. I will wait to see what fruits the pending discussions bear.

Reply:
I wasn't sure if/when this would next be looked at, but I didn't want to leave the code is a bad place. I went ahead and applied the updates you suggested and implemented a way to dispose the CancellationTokenRegistration once its no longer needed.